src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of Bounds");
+        }
+
+        if (size == elementData.length) {
+            grow();
+        }
 
+        System.arraycopy(elementData, index, elementData, index + 1,
+                elementData.length - (index + 1));
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            if (size == elementData.length) {
+                grow();
+            }
+            elementData[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (checkIndex(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Index Out Of Bounds");
+        }
+        return elementData[index];
```

Comment:
Remove repeating code please
Throw Exception inside checkIndex metod and make Exception msg more informative (add size/index info)
```suggestion
        checkIndex(index);
        return elementData[index];
```